### DISCUSSION TITLE
Interactive and automatic filtering

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#360] [General]
I just applied the tool on the backend of zeeguu and got this:

<img width="1309" alt="image" src="https://user-images.githubusercontent.com/464519/157517734-33c2712f-a68e-44ab-b9ea-f160d2c3851f.png">

the very big yellow blotch is a 1.5M .svg file (that is nothing else, but another visualization of the project generated with graphviz:

https://raw.githubusercontent.com/zeeguu-ecosystem/zeeguu-api/64be8fea7df4b91209dcc80471ac9f0e7a66e64d/zeeguu.svg

lessons learned:
- it's good that you're not drawing dependencies between the nodes :)
- we need to figure out a way of removing such files from the view 
   - interactive filtering as we discussed today (ideally also with saving in a .truckignore)
   - making the size metric only work for "code" files... 
   - the problem might be mitigated in this particular case when we'll allow "zooming-in" in the zeeguu folder; however, this would not work if the big files were somewhere deep in the folder tree

on the bright side, the treemap view looks a bit like a rotated flag of ukraine, so that's nice: 

<img width="2115" alt="image" src="https://user-images.githubusercontent.com/464519/157519205-f718cb4a-c76b-457e-9c27-d3d488e3f67f.png">

- [x] #331
- [x] #348
- [x] #346

**Comment by nimrossum** [comment]
> interactive filtering as we discussed today (ideally also with saving in a .truckignore)

This is tracked in #331. 

> making the size metric only work for "code" files...

Tracked in #348

> the problem might be mitigated in this particular case when we'll allow "zooming-in" in the zeeguu folder; however, this would not work if the big files were somewhere deep in the folder tree

Tracked in #346
